portfolio/src/main/java/com/google/sps/servlets/BasketballCourtsServlet.java

```diff
@@ -0,0 +1,29 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.BasketballCourtsDatastore;
+import com.google.sps.data.BasketballCourt;
+
+/** Returns the accessible basketball courts in New York City */
+@WebServlet("/basketball-courts")
+public final class BasketballCourtsServlet extends HttpServlet {
+  List<BasketballCourt> accessibleCourts;
+
+  @Override
+  public void init() {
+    accessibleCourts = BasketballCourtsDatastore.getAccessibleBasketballCourts();
```

Comment:
as mentioned in the BasketballCourtsDataStore file, the caching of your File access is a responsibility of your datastore. For all you know your datastore could be reading from database. so the servlet should not cache that value. The datastore knows the source is a static file so cacheable so it shoudl manage the caching strategy inside the class itself not delegate it. If multiple servlets are calling the getAccessibleCourts they will all read from file instead of hitting the cache as they should.